mongodbforms/documents.py

```diff
@@ -116,7 +116,21 @@ def construct_instance(form, instance, fields=None, exclude=None):
         elif isinstance(f, ListField):
             list_field = getattr(instance, f.name)
             uploads = cleaned_data[f.name]
+            idx_to_pop = []
             for i, uploaded_file in enumerate(uploads):
+                if isinstance(uploaded_file, list):  # ListOfFilesWidget
```

Comment:
Can you test instead on `isinstance(f, ListOfFilesWidget)` as is done above? Not a huge fan of this, seems like dark magic, and in reality makes the `to_delete` aspect a bit more complicated than it perhaps needs to be?